csharp/src/ProtocolBuffers/CodedOutputStream.cs

```diff
@@ -534,178 +443,13 @@ public void WriteRawTag(byte b1, byte b2, byte b3, byte b4, byte b5)
         }
         #endregion
 
-        #region Write packed array members
-        // TODO(jonskeet): A lot of these are really inefficient, due to method group conversions. Fix!
-        // (Alternatively, add extension methods to RepeatedField, accepting the Write* methods via delegates too.)
-        public void WritePackedBoolArray(RepeatedField<bool> list)
-        {
-            uint size = (uint)list.Count;
-            WriteRawVarint32(size);
-            foreach (var value in list)

```

Comment:
We've found out that foreach can be slow, right? for(int i = 0; ....) might be better here (and below).